onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -4409,16 +4409,37 @@ def aten_sinh(self: TFloat) -> TFloat:
     return op.Sinh(self)
 
 
+@torch_op("aten::slice")
 def aten_slice(
-    self: TensorType,
+    self: TTensor,
     dim: int = 0,
     start: Optional[INT64] = None,
     end: Optional[INT64] = None,
     step: INT64 = 1,
-) -> TensorType:
+) -> TTensor:
     # slice.Tensor(Tensor(a) self, int dim=0, SymInt? start=None, SymInt? end=None, SymInt step=1) -> Tensor(a)
 
-    raise NotImplementedError()
+    is_scalar = op.Size(op.Shape(start)) == 0
```

Comment:
Since start and end are Optional, we should leverage OptionalHasElement/OptionalGetElement to get their values.

Reply:
add TODO to implementation later due to HasElement() cannot work now.